### PR TITLE
CLOUDP-101199: Improve Flakiness of Cloud Manager tests in MongoCLI

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes
Improve Flakiness of Cloud Manager tests in MongoCLI by creating a new project id for each test run

_Jira ticket:_ CLOUDP-101199

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](e2e/E2E-TESTS.md) (if a new command or e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by andreaangiolillo — APPROVED** [review]
LGTM. Great work 💯

File: build/ci/evergreen.yml [hunk 1]
```diff
@@ -175,6 +175,39 @@ functions:
...
+          MCLI_PUBLIC_API_KEY: ${cloud_manager_public_api_key}
+          MCLI_OPS_MANAGER_URL: ${mcli_ops_manager_url}
+          MCLI_SERVICE: cloud-manager
+  "clean-up cloud manager":
+    - command: subprocess.exec
+      type: setup
+      params:
+        add_to_path:
+          - ${go_bin}
+          - ../../bin
```
Comment (threebee) [inline build/ci/evergreen.yml:205]: [Question for my own learning] Why are we using relative path like this `../../bin` and not relative with respect to `${workdir}/`?

**Review by threebee — APPROVED** [review]
LGTM, I have added a question.

File: build/ci/evergreen.yml (hunk 1 above)
Comment (fmenezes) [reply build/ci/evergreen.yml:205]: the `../../` is relative to `workdir` that is configured to be `src/github.com/mongodb/mongocli/build/ci`, in other words, resulting destination will be `src/github.com/mongodb/mongocli/bin`, I've tried the path straight away and for some reason it did not work.

**Pull request merged**